examples/import-database-flyway/README.adoc

```diff
@@ -0,0 +1,223 @@
+= Import Database By Flyway
+:index-group: Import
+:jbake-type: page
+:jbake-status: not published/unrevised
+
+This is an example of how to use the Flyway tool to import a database programmatically.
+
+[discrete]
+==== Import Database By Flyway
+
+To use Flyway, at application runtime within the application server, we will use the application server's DataSource, but for unit testing/integrated testing, we will use a configured DataSource to use an in-memory database.
+
+[source,java]
+----
+public class ImportByFlyway {
+	private static final Logger LOGGER = Logger.getInstance(LogCategory.OPENEJB,
+			EntityManagerFactoryCallable.class.getName());
+
+	public static final String IMPORT_FILE_PREFIX = "V_";
+	public static final String IMPORT_FILE_EXTENSION = ".sql";
+
+	private final DataSource dataSource;
+	private final String resource;
+
+	public ImportByFlyway(final ClassLoader cl, final String resource, final DataSource ds) {
+		this.dataSource = ds;
+		this.resource = resource;
+
+		if (dataSource == null) {
+			throw new NullPointerException("datasource can't be null");
+		}
+
+	}
+
+	public void doImport() {
+
+		try {
+
+			try {
+				
+				List<String> sqlFiles = listFilteredFiles(resource, Integer.MAX_VALUE);
+
+				if (Objects.nonNull(sqlFiles)) {
+					if (sqlFiles.isEmpty()) {
+						LOGGER.error("The Resource directory for sql files, can not to be empty.");
+						throw new Exception("The Resource directory for sql files, can not to be empty.");
+					}
+				}
+
+			} catch (final IOException e) {
+				throw new OpenEJBRuntimeException("The Resource directory for sql files, can not to be empty.", e);
+			}
+
+			Flyway flyway = Flyway.configure().locations("filesystem:src/test/resources").dataSource(dataSource)
+					.cleanDisabled(false).load();
+
+			flyway.clean();
```

Comment:
Might not be a good idea to suggest to use `clean` ;-) - if you do that in a productive app, you need to restore from backup.

Reply:
Ok, i will change.